### PR TITLE
refactor: webpack config

### PR DESCRIPTION
**Pull request by devCrossNet**
### What is accomplished by your PR?
This PR hopefully simplifies the way we manage the webpack configs.

### Is there something controversial in your PR?
not that I know

### Link to the Issue
no ticket

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your PR
- [ ] Add new documentation for the code introduced by your PR

### CODE REVIEW

File: docs/guide/third-party.md [hunk 1]
```diff
@@ -3,11 +3,13 @@
 ## Vuetify
 
 ### Installation
+
 Before we start please install vuetify and the webpack css-loader by running `npm install vuetify --save --save-exact && npm install --save-dev --save-exact css-loader` and follow the [Clean-Up-Guide](/guide/clean-up.md).
 
 ### Configure Webpack
 
-`./config/webpack.base.config.js`:
+`./config/base.js`:
```
Comment (tamer-mohamed) [inline docs/guide/third-party.md]: how about renaming `config` folder to `webpack`? more explicit.

File: config/base-server.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const devServer = utils.merge(base, {
+  target: 'node',
+  output: {
+    path: utils.resolve('dist/server'),
+    libraryTarget: 'commonjs',
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      CLIENT: false,
```
Comment (tamer-mohamed) [inline config/base-server.js]: We can use one or the other, `CLIENT: true | false` and get rid of `SERVER`.
`CLIENT` is enough to determine the context whether it is client or server, IMO.

File: docs/guide/third-party.md (hunk 1 above)
Comment (devCrossNet) [reply docs/guide/third-party.md]: I like it 👍

**Comment by devCrossNet** [comment]
@tamer-mohamed I moved the whole webpack thing to the tools folder (we could potentially break this out into its own package later). I also changed the webpack-configs to typescript. 

Can you do me the favour to test this PR before I merge it? (dev, build, test, etc.)

**Comment by tamer-mohamed** [comment]
@devCrossNet `build`  `dev` and `test` tasks works as it should.

About the packages, I have a proposal for the packages and working solution, we should discuss it soon.

**Pull request merged**